File: MergeToSurvive/src/com/vgdc/merge/entities/abilities/ProjectileAbility.java
package com.vgdc.merge.entities.abilities;

import com.badlogic.gdx.math.Vector2;
import com.vgdc.merge.entities.Entity;
import com.vgdc.merge.entities.EntityData;

public class ProjectileAbility extends Ability {
	
	public String projectile;
	
	public ProjectileAbility()
	{
//		
	}
	
	public ProjectileAbility(String projectile)
	{
		this.projectile = projectile;
	}

	@Override
	public void onUse(Entity entity, boolean retrievable) {
		EntityData data = entity.getWorld().getHandler().getEntityData(projectile);
		Entity projectile = createProjectile(data, entity, retrievable);
		projectile.getMovingBody().setVelocity(new Vector2((projectile.getRenderer().isFlipped()? -data.moveSpeed : data.moveSpeed),data.jumpHeight));
	}
	
	

}

File: MergeToSurvive-android/assets/test/scripts/ProjectileAbility.py
from com.vgdc.merge.entities.abilities import Ability
from com.badlogic.gdx.math import Vector2
from com.vgdc.merge.entities import Entity
from com.vgdc.merge.entities import EntityData;

class ProjectileAbility(Ability):
	
	def __init__(self, projectile):
		self.projectile = projectile

	def onUse(self, entity, retrievable = False):
		data = entity.getWorld().getHandler().getEntityData(self.projectile)
		projectile = self.createProjectile(data, entity, retrievable)
		moveSpeed = data.moveSpeed
		if projectile.facingLeft():
			moveSpeed = -moveSpeed
		projectile.getMovingBody().setVelocity(Vector2(moveSpeed,data.jumpHeight))
